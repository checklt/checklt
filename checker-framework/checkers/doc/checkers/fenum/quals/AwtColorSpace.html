<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:46 PDT 2013 -->
<title>AwtColorSpace</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AwtColorSpace";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../checkers/fenum/quals/AwtAlphaCompositingRule.html" title="annotation in checkers.fenum.quals"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../checkers/fenum/quals/AwtCursorType.html" title="annotation in checkers.fenum.quals"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?checkers/fenum/quals/AwtColorSpace.html" target="_top">Frames</a></li>
<li><a href="AwtColorSpace.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Required&nbsp;|&nbsp;</li>
<li>Optional</li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Element</li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">checkers.fenum.quals</div>
<h2 title="Annotation Type AwtColorSpace" class="title">Annotation Type AwtColorSpace</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre><a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Documented.html?is-external=true" title="class or interface in java.lang.annotation">@Documented</a>
 <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Retention.html?is-external=true" title="class or interface in java.lang.annotation">@Retention</a>(<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Retention.html?is-external=true#value()" title="class or interface in java.lang.annotation">value</a>=<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/RetentionPolicy.html?is-external=true#RUNTIME" title="class or interface in java.lang.annotation">RUNTIME</a>)
 <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Target.html?is-external=true" title="class or interface in java.lang.annotation">@Target</a>(<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Target.html?is-external=true#value()" title="class or interface in java.lang.annotation">value</a>={<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/ElementType.html?is-external=true#TYPE_USE" title="class or interface in java.lang.annotation">TYPE_USE</a>,<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/ElementType.html?is-external=true#TYPE_PARAMETER" title="class or interface in java.lang.annotation">TYPE_PARAMETER</a>})
 <a href="../../../checkers/quals/TypeQualifier.html" title="annotation in checkers.quals">@TypeQualifier</a>
 <a href="../../../checkers/quals/SubtypeOf.html" title="annotation in checkers.quals">@SubtypeOf</a>(<a href="../../../checkers/quals/SubtypeOf.html#value()">value</a>=<a href="../../../checkers/fenum/quals/FenumTop.html" title="annotation in checkers.fenum.quals">FenumTop.class</a>)
public @interface <span class="strong">AwtColorSpace</span></pre>
<div class="block">Color space tags to identify the specific color space of a Color
 object or, via a ColorModel object, of an Image, a BufferedImage,
 or a GraphicsDevice (see <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true" title="class or interface in java.awt.color"><code>ColorSpace</code></a> for
 more details).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_XYZ" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_XYZ</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_Lab" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_Lab</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_Luv" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_Luv</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_YCbCr" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_YCbCr</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_Yxy" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_Yxy</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_RGB" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_RGB</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_GRAY" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_GRAY</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_HSV" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_HSV</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_HLS" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_HLS</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_CMYK" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_CMYK</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_CMY" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_CMY</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_2CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_2CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_3CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_3CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_4CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_4CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_5CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_5CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_6CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_6CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_7CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_7CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_8CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_8CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_9CLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_9CLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_ACLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_ACLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_BCLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_BCLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_CCLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_CCLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_DCLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_DCLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_ECLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_ECLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#TYPE_FCLR" title="class or interface in java.awt.color"><code>ColorSpace.TYPE_FCLR</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#CS_sRGB" title="class or interface in java.awt.color"><code>ColorSpace.CS_sRGB</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#CS_LINEAR_RGB" title="class or interface in java.awt.color"><code>ColorSpace.CS_LINEAR_RGB</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#CS_CIEXYZ" title="class or interface in java.awt.color"><code>ColorSpace.CS_CIEXYZ</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#CS_PYCC" title="class or interface in java.awt.color"><code>ColorSpace.CS_PYCC</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/color/ColorSpace.html?is-external=true#CS_GRAY" title="class or interface in java.awt.color"><code>ColorSpace.CS_GRAY</code></a></dd></dl>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../checkers/fenum/quals/AwtAlphaCompositingRule.html" title="annotation in checkers.fenum.quals"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../checkers/fenum/quals/AwtCursorType.html" title="annotation in checkers.fenum.quals"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?checkers/fenum/quals/AwtColorSpace.html" target="_top">Frames</a></li>
<li><a href="AwtColorSpace.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Required&nbsp;|&nbsp;</li>
<li>Optional</li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Element</li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
