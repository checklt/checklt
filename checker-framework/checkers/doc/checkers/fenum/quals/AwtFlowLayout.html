<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:46 PDT 2013 -->
<title>AwtFlowLayout</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AwtFlowLayout";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../checkers/fenum/quals/AwtCursorType.html" title="annotation in checkers.fenum.quals"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../checkers/fenum/quals/Fenum.html" title="annotation in checkers.fenum.quals"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?checkers/fenum/quals/AwtFlowLayout.html" target="_top">Frames</a></li>
<li><a href="AwtFlowLayout.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Required&nbsp;|&nbsp;</li>
<li>Optional</li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Element</li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">checkers.fenum.quals</div>
<h2 title="Annotation Type AwtFlowLayout" class="title">Annotation Type AwtFlowLayout</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre><a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Documented.html?is-external=true" title="class or interface in java.lang.annotation">@Documented</a>
 <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Retention.html?is-external=true" title="class or interface in java.lang.annotation">@Retention</a>(<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Retention.html?is-external=true#value()" title="class or interface in java.lang.annotation">value</a>=<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/RetentionPolicy.html?is-external=true#RUNTIME" title="class or interface in java.lang.annotation">RUNTIME</a>)
 <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Target.html?is-external=true" title="class or interface in java.lang.annotation">@Target</a>(<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/Target.html?is-external=true#value()" title="class or interface in java.lang.annotation">value</a>={<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/ElementType.html?is-external=true#TYPE_USE" title="class or interface in java.lang.annotation">TYPE_USE</a>,<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/lang/annotation/ElementType.html?is-external=true#TYPE_PARAMETER" title="class or interface in java.lang.annotation">TYPE_PARAMETER</a>})
 <a href="../../../checkers/quals/TypeQualifier.html" title="annotation in checkers.quals">@TypeQualifier</a>
 <a href="../../../checkers/quals/SubtypeOf.html" title="annotation in checkers.quals">@SubtypeOf</a>(<a href="../../../checkers/quals/SubtypeOf.html#value()">value</a>=<a href="../../../checkers/fenum/quals/FenumTop.html" title="annotation in checkers.fenum.quals">FenumTop.class</a>)
public @interface <span class="strong">AwtFlowLayout</span></pre>
<div class="block">Line alignments in a flow layout (see
 <a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true" title="class or interface in java.awt"><code>FlowLayout</code></a> for more details).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true#LEFT" title="class or interface in java.awt"><code>FlowLayout.LEFT</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true#CENTER" title="class or interface in java.awt"><code>FlowLayout.CENTER</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true#RIGHT" title="class or interface in java.awt"><code>FlowLayout.RIGHT</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true#LEADING" title="class or interface in java.awt"><code>FlowLayout.LEADING</code></a>, 
<a href="http://types.cs.washington.edu/checker-framework/api/jdk/java/awt/FlowLayout.html?is-external=true#TRAILING" title="class or interface in java.awt"><code>FlowLayout.TRAILING</code></a></dd></dl>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../checkers/fenum/quals/AwtCursorType.html" title="annotation in checkers.fenum.quals"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../checkers/fenum/quals/Fenum.html" title="annotation in checkers.fenum.quals"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?checkers/fenum/quals/AwtFlowLayout.html" target="_top">Frames</a></li>
<li><a href="AwtFlowLayout.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Required&nbsp;|&nbsp;</li>
<li>Optional</li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Element</li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
