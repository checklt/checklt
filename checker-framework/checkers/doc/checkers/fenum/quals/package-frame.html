<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:48 PDT 2013 -->
<title>checkers.fenum.quals</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../../checkers/fenum/quals/package-summary.html" target="classFrame">checkers.fenum.quals</a></h1>
<div class="indexContainer">
<h2 title="Annotation Types">Annotation Types</h2>
<ul title="Annotation Types">
<li><a href="AwtAlphaCompositingRule.html" title="annotation in checkers.fenum.quals" target="classFrame">AwtAlphaCompositingRule</a></li>
<li><a href="AwtColorSpace.html" title="annotation in checkers.fenum.quals" target="classFrame">AwtColorSpace</a></li>
<li><a href="AwtCursorType.html" title="annotation in checkers.fenum.quals" target="classFrame">AwtCursorType</a></li>
<li><a href="AwtFlowLayout.html" title="annotation in checkers.fenum.quals" target="classFrame">AwtFlowLayout</a></li>
<li><a href="Fenum.html" title="annotation in checkers.fenum.quals" target="classFrame">Fenum</a></li>
<li><a href="FenumTop.html" title="annotation in checkers.fenum.quals" target="classFrame">FenumTop</a></li>
<li><a href="FenumUnqualified.html" title="annotation in checkers.fenum.quals" target="classFrame">FenumUnqualified</a></li>
<li><a href="SwingBoxOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingBoxOrientation</a></li>
<li><a href="SwingCompassDirection.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingCompassDirection</a></li>
<li><a href="SwingElementOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingElementOrientation</a></li>
<li><a href="SwingHorizontalOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingHorizontalOrientation</a></li>
<li><a href="SwingSplitPaneOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingSplitPaneOrientation</a></li>
<li><a href="SwingTextOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingTextOrientation</a></li>
<li><a href="SwingTitleJustification.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingTitleJustification</a></li>
<li><a href="SwingTitlePosition.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingTitlePosition</a></li>
<li><a href="SwingVerticalOrientation.html" title="annotation in checkers.fenum.quals" target="classFrame">SwingVerticalOrientation</a></li>
</ul>
</div>
</body>
</html>
