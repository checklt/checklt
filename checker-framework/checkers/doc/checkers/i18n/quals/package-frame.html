<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:48 PDT 2013 -->
<title>checkers.i18n.quals</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../../checkers/i18n/quals/package-summary.html" target="classFrame">checkers.i18n.quals</a></h1>
<div class="indexContainer">
<h2 title="Annotation Types">Annotation Types</h2>
<ul title="Annotation Types">
<li><a href="LocalizableKey.html" title="annotation in checkers.i18n.quals" target="classFrame">LocalizableKey</a></li>
<li><a href="Localized.html" title="annotation in checkers.i18n.quals" target="classFrame">Localized</a></li>
</ul>
</div>
</body>
</html>
