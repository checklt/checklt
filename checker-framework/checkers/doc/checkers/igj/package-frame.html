<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:48 PDT 2013 -->
<title>checkers.igj</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../checkers/igj/package-summary.html" target="classFrame">checkers.igj</a></h1>
<div class="indexContainer">
<h2 title="Classes">Classes</h2>
<ul title="Classes">
<li><a href="IGJAnnotatedTypeFactory.html" title="class in checkers.igj" target="classFrame">IGJAnnotatedTypeFactory</a></li>
<li><a href="IGJChecker.html" title="class in checkers.igj" target="classFrame">IGJChecker</a></li>
<li><a href="IGJVisitor.html" title="class in checkers.igj" target="classFrame">IGJVisitor</a></li>
</ul>
</div>
</body>
</html>
