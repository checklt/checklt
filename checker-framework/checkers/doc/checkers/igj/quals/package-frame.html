<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.7.0_01) on Wed May 01 23:33:48 PDT 2013 -->
<title>checkers.igj.quals</title>
<meta name="date" content="2013-05-01">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../../checkers/igj/quals/package-summary.html" target="classFrame">checkers.igj.quals</a></h1>
<div class="indexContainer">
<h2 title="Annotation Types">Annotation Types</h2>
<ul title="Annotation Types">
<li><a href="Assignable.html" title="annotation in checkers.igj.quals" target="classFrame">Assignable</a></li>
<li><a href="AssignsFields.html" title="annotation in checkers.igj.quals" target="classFrame">AssignsFields</a></li>
<li><a href="I.html" title="annotation in checkers.igj.quals" target="classFrame">I</a></li>
<li><a href="Immutable.html" title="annotation in checkers.igj.quals" target="classFrame">Immutable</a></li>
<li><a href="Mutable.html" title="annotation in checkers.igj.quals" target="classFrame">Mutable</a></li>
<li><a href="ReadOnly.html" title="annotation in checkers.igj.quals" target="classFrame">ReadOnly</a></li>
</ul>
</div>
</body>
</html>
