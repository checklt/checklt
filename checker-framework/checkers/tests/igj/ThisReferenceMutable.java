import checkers.igj.quals.*;

@I
class ThisReferenceMutableSuper {

    // Receiving methods
    public static void isRO(@ReadOnly ThisReferenceMutableSuper obj) { }
    public static void isMutable(@Mutable ThisReferenceMutableSuper obj) { }
    public static void isImmutable(@Immutable ThisReferenceMutableSuper obj) { }

}

class MutableThisTest extends ThisReferenceMutableSuper {
    public void testRO(@ReadOnly MutableThisTest this) {
        isRO(this);
        isMutable(this);   // should emit error
        isImmutable(this); // should emit error
    }

    public void testAssignsFields (@AssignsFields MutableThisTest this) {
        isRO(this);
        isMutable(this);   // should emit error
        isImmutable(this); // should emit error
    }

    public void testMutable(@Mutable MutableThisTest this) {
        isRO(this);
        isMutable(this);
        isImmutable(this); // should emit error
    }
}
