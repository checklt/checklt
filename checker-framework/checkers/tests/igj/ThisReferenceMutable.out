ThisReference.java:16: error: (argument.type.incompatible)
ThisReference.java:17: error: (argument.type.incompatible)
ThisReference.java:22: error: (argument.type.incompatible)
ThisReference.java:23: error: (argument.type.incompatible)
ThisReference.java:29: error: (argument.type.incompatible)
